### ISSUE TITLE
Emoji do not work on iOS 😦

### ISSUE THREAD

**Issue by eseidelGoogle** [flutter/flutter#4273]
😦 

So Chrome had this bug.  https://bugs.chromium.org/p/chromium/issues/detail?id=62435 We forked our Text system from Chrome before they fixed it.  So presumably it's just a few patches to pull.  Skia supports Emoji on Mac now, presumably also on iOS too.  FYI @mikerreed @abarth @chinmaygarde 

Flutter main.dart:

```
import 'package:flutter/material.dart';

void main() {
  runApp(
    new MaterialApp(
      title: 'Emoji Demo 👍',
      theme: new ThemeData(
        primarySwatch: Colors.blue
      ),
      home: new Scaffold(
        appBar: new AppBar(
          title: new Text('Emoji Demo 👍')
        ),
        body: new Center(
          child: new Text('😂')
        )
      )
    )
  );
}
```

Android device (Nexus 5):
![flutter_01](https://cloud.githubusercontent.com/assets/11857803/15682138/20341d78-2711-11e6-8b17-3fac8567b9b2.png)

iOS Simulator (iPhone 6):
![simulator screen shot may 31 2016 9 22 46 am](https://cloud.githubusercontent.com/assets/11857803/15682164/40003538-2711-11e6-9818-97a6b4b92677.png)

**Comment by abarth** [comment]
Maybe all we need to do is merge https://src.chromium.org/viewvc/blink/trunk/Source/platform/fonts/mac/FontCacheMac.mm?r1=186991&r2=186990&pathrev=186991 ?

**Comment by abarth** [comment]
Hum...  I don't actually see that blacklist in our code.

**Comment by sethladd** [comment]
@chinmaygarde were you working on this?

**Comment by chinmaygarde** [comment]
Yup. Trying to work my way through getting the Apple Color Emoji to render.

**Comment by eseidelGoogle** [comment]
Chinmay is close.  But I just spoke with him and he's willing to pause this in order to let us have https://github.com/flutter/flutter/issues/3805 done in time for our next demos in the coming weeks.

**Comment by abarth** [comment]
We should probably pick this up again.  It sounds like we were close a while back.

**Comment by jtrunick** [comment]
If you don't mind checking ♭works, I'd appreciate it.

**Comment by xster** [comment]
Fixed along with #6939 via https://github.com/flutter/engine/issues/3418
